File: selenium-google/google.py
#!/usr/bin/env python3
import time
import selenium
from selenium import webdriver
from selenium.webdriver.common.by import By

def edgeDriver():
    options = webdriver.EdgeOptions()
    options.add_argument("--disable-blink-features=AutomationControlled")
    options.add_experimental_option("excludeSwitches", ["enable-automation", "enable-logging"])

    driver = webdriver.Edge(options)
    driver.set_page_load_timeout(30)
    return driver

def exec(driver):
    driver.get('https://www.google.com/')
    driver.find_element(By.ID, "APjFqb").send_keys("selenium")
    driver.find_elements(By.NAME, "btnK")[1].click()
    print('--- executed ---')

def main():
    driver = edgeDriver()
    exec(driver)
    time.sleep(10)
    driver.quit()

if __name__ == "__main__":
    main()

File: selenium-google/Google.java
import java.time.Duration;
import java.util.List;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.edge.EdgeDriver;
import org.openqa.selenium.edge.EdgeOptions;

public class Google {

    private static WebDriver edgeDriver() {
        var options = new EdgeOptions();
        options.addArguments("--disable-blink-features=AutomationControlled");
        options.setExperimentalOption("excludeSwitches", List.of("enable-automation", "enable-logging"));

        var driver = new EdgeDriver(options);
        driver.manage().timeouts().pageLoadTimeout(Duration.ofSeconds(30));
        return driver;
    }

    private static void exec(WebDriver driver) {
        driver.get("https://www.google.com/");
        driver.findElement(By.id("APjFqb")).sendKeys("selenium");
        driver.findElements(By.name("btnK")).get(1).click();
        System.out.println("--- executed ---");
    }

    public static void main(String[] args) throws Exception {
        var driver = edgeDriver();
        exec(driver);
        Thread.sleep(10_000L);
        driver.quit();
    }
}
